snapcraft.yaml

```diff
@@ -101,3 +101,84 @@ parts:
       snapcraftctl prime
       # ensure build-in tests are run
       cd ${SNAPCRAFT_PART_SRC} && make test TESTDIR=${SNAPCRAFT_PRIME}
+
+  opensc:
+    plugin: autotools
+    source: https://git.launchpad.net/ubuntu/+source/opensc
+    source-type: git
+    source-depth: 1
+    source-branch: applied/ubuntu/focal
+    autotools-configure-parameters:
+      - --host ${SNAPCRAFT_ARCH_TRIPLET}
+      - --prefix=/usr
+      - --sysconfdir=/etc/opensc
+      - --with-pkcs11-provider=/usr/lib/${SNAPCRAFT_ARCH_TRIPLET}/pkcs11/p11-kit-client.so
+      - --libdir=/usr/lib/${SNAPCRAFT_ARCH_TRIPLET}
+      - --disable-man
+      - --disable-sm
+    build-environment:
+      - LDFLAGS: "-w -s"
+    override-build: |
+      snapcraftctl build
+      make install-strip DESTDIR=${SNAPCRAFT_PART_INSTALL}
+    stage:
+      - usr/bin/pkcs11-tool
+      - usr/lib/*/libopensc.so*
+
+  opensc-libp11:
+    plugin: autotools
+    source: https://git.launchpad.net/ubuntu/+source/libp11
+    source-type: git
+    source-depth: 1
+    source-branch: applied/ubuntu/focal
+    autotools-configure-parameters:
+      - --host ${SNAPCRAFT_ARCH_TRIPLET}
+      - --prefix=/usr
+      - --sysconfdir=/etc/opensc
+      - --with-pkcs11-module=/usr/lib/${SNAPCRAFT_ARCH_TRIPLET}/pkcs11/p11-kit-client.so
+      - --libdir=/usr/lib/${SNAPCRAFT_ARCH_TRIPLET}
+    build-environment:
+      - LDFLAGS: "-w -s"
+    override-build: |
+      snapcraftctl build
+      make install-strip DESTDIR=${SNAPCRAFT_PART_INSTALL}
+    stage:
+      - usr/lib/*/*.so*
+      - usr/lib/*/*/*.so*
+
+  p11:
+    plugin: nil
+    stage-packages:
+      # libp11-kit0 is already included
+      # libtasn1-6 is already included
+      - gnutls-bin
+      - libopts25
+      - p11-kit
+      - p11-kit-modules
+    override-build: |
+      mkdir -p ${SNAPCRAFT_PART_INSTALL}/etc/pkcs11/modules
+      echo -e "module: /usr/lib/${SNAPCRAFT_ARCH_TRIPLET}/pkcs11/p11-kit-client.so" > \
+              ${SNAPCRAFT_PART_INSTALL}/etc/pkcs11/modules/p11-kit-client.module
+    stage:
+      - etc
+      - usr/bin/p11tool
+      - usr/bin/p11-kit
+      - usr/libexec
+      - usr/lib/${SNAPCRAFT_ARCH_TRIPLET}/libopts.so*
+      - usr/lib/${SNAPCRAFT_ARCH_TRIPLET}/p11-kit-proxy.so*
+      - usr/lib/${SNAPCRAFT_ARCH_TRIPLET}/pkcs11/p11-kit-client.so
+
+build-packages:
```

Comment:
this seems like it is meant to be a part of the `p11` part?

Reply:
do you mean 
`build-packages:`
no, you can define `build-packages:` in top level, rather for each part and repeat same dependency multiple times.
I assume this is more matter of preferences though...

Reply:
oh wow TIL top level `build-packages` is a thing I had no idea, always learning :smile: